File: 1984-maximum-distance-between-a-pair-of-values/maximum-distance-between-a-pair-of-values.cpp
class Solution {
public:
    int maxDistance(vector<int>& nums1, vector<int>& nums2) {
        int left = 0;
        int right = 0;
        int ans = 0;
        while(left<nums1.size() && right<nums2.size()){
            // if(left > right){
            //     right++;
            //     continue;

            // }
            if(nums1[left]<=nums2[right]){
                ans = max(ans,right-left);
                right++;

            }
            else{
                left++;
            }
        }
        return ans;
        
    }
};

/*
If nums1[i] <= nums2[j]:

This is a valid pair
Update answer with j - i
Move j forward to try for a larger distance...other wise its invalid pair */

File: 1984-maximum-distance-between-a-pair-of-values/maximum-distance-between-a-pair-of-values.java
class Solution {
    public int maxDistance(int[] arr1, int[] arr2) {
        int i = 0, j = 0;
        int n = arr1.length, m = arr2.length;
        int mx = 0;

        while (i < n && j < m) {

            if (arr1[i] <= arr2[j]) {
                mx = Math.max(mx, j - i);
                j++;
            } else {
                i++;
            }
        }

        return mx;
    }
}